README.md

```diff
@@ -106,161 +136,183 @@ try {
-txParams.put("data", null);
-txParams.put("nrg", "0x989680");
-txParams.put("nrgPrice", "0x989680");
-
-// Create and sign your Transaction object
-Transaction transaction = pocketAion.createTransaction(senderWallet, "32", txParams);
-```
+6- Value
 
-To send your newly created `Transaction` to the node use the `sendTransaction` method:
+After we meet the params, we can now call the `executeFunction`:
 
 ```
+List<Object> functionParams = new ArrayList<>();
+functionParams.add(new BigInteger("1"));
+BigInteger nrg = new BigInteger("50000");
+BigInteger nrgPrice = new BigInteger("10000000000");
+BigInteger nonce = new BigInteger("1");
+BigInteger value = new BigInteger("0");
 try {
-   Transaction response = pocketAion.sendTransaction(transaction);
-} catch(JSONException e) {
-    e.printStackTrace();
-} catch (IOException e) {
-    e.printStackTrace();
+    aionContract.executeFunction("addToState", wallet, functionParams, nonce, nrg, nrgPrice, value, new RPCCallback<String>() {
+        @Override
+        public void onResult(String result, Exception exception) {
+            // The result will contain the transaction hash
+            // The exception, if any, will contain the error sent from the Pocket Node
+        }
+    });
+} catch (CreateTransactionException cte) {
+    cte.printStackTrace();
+} catch (CreateQueryException cqe) {
+    cqe.printStackTrace();
+} catch (AionContractException ace) {
+    ace.printStackTrace();
 }
 ```
 
-## Creating and sending a Query
-Follow the example below to create a `Transaction` object for writing to the given AION network with the parameters below and `subnetwork`. 
-Throws `CreateQueryException` in case of errors.
+### Calling an AionContract function
+There are 2 main distinctions when calling a smart contract function: whether or not calling it alters 
+the state of the smart contract. This is indicated in the `constant` attribute of the JSON.
 
-```
-String[] rpcParams = {"0x0", "latest"};
-Map<String, Object> params = new HashMap<>();
-params.put("rpcMethod", "eth_getBalance");
-params.put("rpcParams", Arrays.asList(rpcParams));
-Query query = this.pocketAion.createQuery("32", params, null);
-```
+To read a smart contract you will need to use the executeConstantFunction call with the following parameters, see the example below:
+
+1- Function name
+
+2- Function parameters
 
-To send your `Query` to the node use the `executeQuery` method:
+3- From address(optional)
 
+4- NRG (Energy, optional)
+
+5- NRG Price (Energy Price, optional)
+
+6- Value (optional)
 ```
+// Solidity function we're calling: function multiply(uint128 a, uint128 b) public constant returns (uint128)
 try {
-   Query response = pocketAion.executeQuery(query);
-} catch(JSONException e) {
-    e.printStackTrace();
-} catch (IOException e) {
-    e.printStackTrace();
+    // Prepare parameters
+    List<Object> functionParams = new ArrayList<>();
+    functionParams.add(new BigInteger("2"));
```

Comment:
Let's remove the functions params, nrg, nrgPrice and from.  Just as in the iOS readme, both tutorials are doing the same just he language changes. send null instead of the fromAddress, nrg, nrgPrice and value.